evalml/pipelines/components/transformers/samplers/base_sampler.py

```diff
@@ -58,6 +60,23 @@ def _prepare_data(self, X, y):
             pd.DataFrame, pd.Series: Prepared X and y data as pandas types
         """
         X = infer_feature_types(X)
+        if len(X.ww.select(IntegerNullable).columns) > 0:
+            try:
+                X = X.astype(
+                    {
+                        null_col: int
+                        for null_col in X.ww.select(IntegerNullable).columns
+                    },
+                )
+            except ValueError:
+                X = X.astype(
+                    {
+                        null_col: float
+                        for null_col in X.ww.select(IntegerNullable).columns
+                    },
+                )
+            X.ww.init(schema=X.ww.schema)
```

Comment:
Keeping `X.ww.init(schema=X.ww.schema)` inside the conditional led to a large improvement in `fit` time